test/chai-sql.coffee

```diff
@@ -133,6 +133,10 @@ exports.operandToAbstractSQLFactory = (binds = [], defaultResource = 'pilot', de
 				mapping = [alias, _.last(fieldParts)]
 			else
 				mapping = [resource, odataNameToSqlName(operand)]
+			# Data type check by field names that are dates
+			if (mapping[1] == 'created at' || mapping[1] == 'modified at' || mapping[1] == 'hire date')
+				return ['DateTrunc', ['EmbeddedText', 'milliseconds'], ['ReferencedField'].concat(mapping)]
+			else
 			return ['ReferencedField'].concat(mapping)
```

Comment:
This should be indented if you want it within the `else`, otherwise you can just drop the `else` altogether and rely on short-circuiting